### PR TITLE
Station rename logging, admin reject button

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
rscadd: Certain custom station names may be rejected by Centcom.
/:cl:
- All renames are logged
- All admin rejections are logged
- A rename generates an admin message which gives them a chance during a window (of 60 seconds) to reject the name
- If not, it changes at the end of that time
- Users can pick a station name that fits the traditional format to skip the waiting period.

### CODE REVIEW

**Comment by lzimann** [comment]
but i want my meme names

**Comment by coiax** [comment]
I think one of the admins said "We should not make it update the name on the hub".
If you don't want the hub seeing your name "FAT SLUT ORGY", maybe you shouldn't pick it?

**Comment by MMMiracles** [comment]
Isn't that the same thing with custom-naming the station to begin with?

Actual fun removal, make admins deal with shits who insist on giant meme names instead.

File: code/game/objects/items/charter.dm [hunk 1]
```diff
@@ -1,24 +1,156 @@
...
+
+	data["can_pick"] = CAN_CHANGE_NAME
+
+	data["potential_name"] = assemble_name()
+	data["current_name"] = world.name
+
+	if(!captain_name)
+		for(var/i in mob_list)
+			var/mob/M = i
+			if(M.mind && M.mind.assigned_role == "Captain")
```
Comment (RemieRichards) [inline code/game/objects/items/charter.dm]: there must be a better way to do this.

**Comment by Aloraydrel** [comment]
No fun allowed

**Comment by KorPhaeron** [comment]
This was an inordinate amount of work to achieve the same results as deleting it

**Comment by coiax** [comment]
I wasn't even trying that hard to come up with interesting station names, I'm sure creative people could find better ones.

And I cannot make admins do things.

What I was considering was giving the custom option, but having it required to be approved by an admin to be accepted. Changes from reactive moderation to proactive. But I wanted to get the code out early anyway, readding custom names with approval won't be difficult.

**Comment by lzimann** [comment]
we can reset the station name at will, so having only custom names is not a problem

**Comment by coiax** [comment]
There is a difference between approving explicitly and improving implicitly. Currently if someone names the station something offensive, and the admin on duty doesn't change it back, there's no record created, either of the rename or the admin. Explicit acceptance will just be as simple as clicking on a link when they propose it, a la centcom messages.

**Comment by lzimann** [comment]
this not needed tbh

**Comment by xxalpha** [comment]
Don't have a feature require admin approval.

**Comment by coiax** [comment]
Like requesting the nuke codes or moving the ferry or TC trades?

**Comment by ChangelingRain** [comment]
This is one of the things you want admin approval on because >all of bee movie as a station name

**Comment by KorPhaeron** [comment]
If they do that then admins have a reset button

>comparing literally nuking the station to giving it s dumb name

**Comment by coiax** [comment]
> There is a difference between approving explicitly and improving implicitly. Currently if someone names the station something offensive, and the admin on duty doesn't change it back, there's no record created, either of the rename or the admin. Explicit acceptance will just be as simple as clicking on a link when they propose it, a la centcom messages.

**Comment by AnturK** [comment]
Why does it need explicit approval though ? This really feels like an overkill (besides the log) just add a reset link next to admin notice. Admin presence shouldn't be necessary for normal play.

**Comment by coiax** [comment]
If admins wanted to not have the selected station name on the hub, what do you think that says about the names that are picked?

And why are admins resisting having to click one link to approve JIZZFEST 2015 as a station name. There's a whole interface for making names that don't require explicit approval, restriction inspiring creativity and all of that. There's a reason this is called fridge magnet.

**Comment by xxalpha** [comment]
>Admin presence shouldn't be necessary for normal play.
>Admin presence shouldn't be necessary for normal play.
>Admin presence shouldn't be necessary for normal play.
>Admin presence shouldn't be necessary for normal play.
>Admin presence shouldn't be necessary for normal play.
>Admin presence shouldn't be necessary for normal play.
>Admin presence shouldn't be necessary for normal play.
>Admin presence shouldn't be necessary for normal play.

**Comment by xxalpha** [comment]
>This is one of the things you want admin approval on because >all of bee movie as a station name

Then limit the input.

**Comment by coiax** [comment]
> renaming the station "Get us off the hub" is normal play now

**Comment by MMMiracles** [comment]
code solutions to problems admins should be dealing with is silly

**Comment by coiax** [comment]
Changes admin oversight from passive to active for something that may be the first impression people get, which is our hub server listing. If the station gets called something offensive while no admins are on, that'll just hang around on the list until the round ends.

In the absence of admins, the limited word list is enough to be creative and distinctive, while vastly reducing the changes of bad.

**Comment by coiax** [comment]
And calling moving approval from implicit to explicit a "code solution" is just wrong. A code solution would be something dumb like screening for naughty words.

**Comment by MMMiracles** [comment]
To be honest I didn't see the changes before making the comment so I just assumed it was still outright denying custom names.

Logging and approval is fine, I would rather see a lack of admin (dis)approval still allow full customization but w/e.

**Comment by Shadowlight213** [comment]
As people seem to be ignoring here, since the name shows up on the hub, this isn't just normal gameplay shit.
Renaming the station has ooc consequences which takes it into the realm of admin approval being required.

**Comment by KorPhaeron** [comment]
>ooc consequences

Has lummox yelled at us for posting bee movie?

**Comment by Firecage** [comment]
I wouldn't mind logging, but a hard restriction like this is fairly silly honestly, considering admins can already reset the name and can already ban shitlers who abuse this, you know, like they ban people who abuse other shit.

And I would prefer naming the station to Zoey's Toy Box even if admins aren't on.

**Comment by Aloraydrel** [comment]
>someone makes a shitty ooc meme name

>gets banned and it changed to a less memey name via admin magic

Its not hard

**Comment by SidonCC** [comment]
Can we just remove the carter, it was a terrible idea

**Comment by optimumtact** [comment]
People do not like this at all

**Comment by optimumtact** [comment]
oh also I forgot to say but I did limit this

**Comment by optimumtact** [comment]
Can you make this a config option that defaults to off please? It's a policy question

**Comment by jughu** [comment]
This is where we add a place to see recent station names by which person

**Comment by MrStonedOne** [comment]
How about just make it be a timed auto approval

Allow admins to reject station names within a 60 second timeframe, tell the player the name is pending "centcom audit" if it is rejected, player can retry.

So minor, so easy and then you can remove all this fridge magnet bs (I know, you like the code, it's your code babby, but its just not gonna work out, (you could make it auto approve custom names that matches))

**Comment by coiax** [comment]
Sounds good.

**Pull request merged**